src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,29 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
```

Comment:
You should save writeLock result to a variable and unlock this var in finally block. Same thing everywhere.